apps/root/src/services/earnService.ts

```diff
@@ -1235,6 +1238,17 @@ export class EarnService extends EventsManager<EarnServiceData> {
     this.userStrategies = userStrategies;
   }
 
+  handleStoredTransaction(transaction: TransactionDetails) {
+    if (!isEarnType(transaction)) return;
+    const existingUserStrategy = this.userStrategies.find((s) => s.id === transaction.typeData.positionId);
+    const isValidCreateTransaction = !existingUserStrategy && transaction.type === TransactionTypes.earnCreate;
+    const isValidTransaction =
+      existingUserStrategy && existingUserStrategy.lastUpdatedAtFromApi < transaction.addedTime;
+    if (!isValidCreateTransaction && !isValidTransaction) return;
```

Comment:
if this is not a valid create transaction but it is a valid transaction it will still go through right?